src/java.base/share/classes/java/lang/invoke/MethodTypeForm.java

```diff
@@ -141,8 +134,6 @@ public LambdaForm cachedLambdaForm(int which) {
         Object entry = lambdaForms[which];
```

Comment:
I recommend restoring the static types of the arrays to `SoftReference<MethodHandle>[]` unless there is extra overhead from loading/initialization of `SoftReference` class.

Reply:
I tried this before but then 'this.methodHandles = new SoftReference<MethodHandle>[MH_LIMIT];' is not allowed due to the error 'generic array creation'.  I looked at allocating using ArrayList<>(initialCapacity) as they are backed. with Object[], but then we have to add range checks on the reads to avoid IndexOutOfBounds or set a null entry at (initialCapacity-1); unless someone can suggest a better implementation I'm inclined to let it stand

Reply:
I restored the code to follow the pre-JEP 483; thank you @liach and @iklam